### PR TITLE
[Countries] Synonyms or possible names for a lot of countries

### LINKED ISSUE #121: Countries EndPoint : countryInfo returns NO DATA for many countries
Hi,

When I hit [https://corona.lmao.ninja/countries](https://corona.lmao.ninja/countries), an array of JSON objects is returned, each element of array is information for one country in JSON.
In each JSON, there is a JSON key-value pair for "countryInfo". This information is returned for most countries. However, it is returned as NO DATA for many countries, Eg. USA, UAE, Syria, etc.
Screenshot from 2020-03-23 22-09-58

### LINKED ISSUE #127: request: Use ISO country names and/or codes
The timeline endpoint (and possibly others) seems to frequently change the names of countries... e.g. it went from `Korea, South` to `korea, south` and then most recently to `s. korea`.

I'm requesting that the standardized (ISO) country names and or 3-letter code be used, to keep everything consistent.  (I will happily provide a JSON of country/state codes, if needed!)

### LINKED ISSUE #128: Some flags come as unkown
Hi there 👋!

Calling this endpoint https://corona.lmao.ninja/countries some flags come with https://raw.githubusercontent.com/NovelCOVID/API/master/assets/flags/unknow.png as value.

For example USA, but it's possible to reach the right flag making a call to: https://raw.githubusercontent.com/NovelCOVID/API/master/assets/flags/us.png

Another example is UK: https://raw.githubusercontent.com/NovelCOVID/API/master/assets/flags/gb.png

Can you update the API for those countries. Thank you in advance and fantastic work! 😊

### LINKED ISSUE #150: countryInfo iso2 and iso3 'NO DATA' for some countries
some countries doesn't have the country ISO codes in the data.

### PR DESCRIPTION
**Pull request by buster95**
Adding synonyms for a lot of countries, homologate 
https://www.worldometers.info/coronavirus/     Country Catalog with this API

This PR has:
- [x] Pairing countries with worldometers web page
- [x] Some spanish country names for searches ex. `Korea del Sur` -> `South Korea`, `Italia` -> `Italy`
- [x] Improvements on countries/{country name} endpoint found countries easier
- [x] Code reduction

This PR
resolve #121 
resolve #127 
resolve #128
resolve #150

### CODE REVIEW

**Comment by EliteDaMyth** [comment]
alot has been changed. have you tested it?

**Comment by buster95** [comment]
sure, was tested.

I wanna know if can I integrate
**`func/countryMap.js`** file into **`utils/country_utils.js`**??
in a next PR

**Comment by buster95** [comment]
@ebwinters you can check this, I modified your code on endpoint **`countries/{country name}`** in file **`server.js`**

**Comment by ebwinters** [comment]
@buster95 why did you remove the standardize countryName code? What is this PR even doing it seems like it's gonna screw a lot of things up

**Comment by buster95** [comment]
@ebwinters 🙄 standardize countryName was moved, check results in my endpoint version and then check results in your endpoint version, after that tell me what version throws better result

**Comment by ebwinters** [comment]
@buster95 I'm tryna fix something for my CLI rn but I'll pull your branch and run locally to test then I'll approve or reject 👍

**Comment by ebwinters** [comment]
I reviewed it and there's a lot of differences for the worse. For example, /countries/korea,%20south should return 
```json
country: "S. Korea"
countryInfo : 
iso2: "NO DATA"
iso3: "NO DATA"
_id: "NO DATA"
lat: 0
long: 0
flag: "https://raw.githubusercontent.com/NovelCOVID/API/master/assets/flags/unknow.png"
cases: 9037
todayCases: 76
deaths: 120
todayDeaths: 9
recovered: 3507
active: 5410
critical: 59
casesPerOneMillion: 176
```
but instead returns
```json
message: "Country not found or dosen't have cases"
```
I'm not sure why this PR is necessary stll, if it's something specific for an application you're building just do it on your end. When I ran it locally a lot of things that worked stopped working

**Comment by buster95** [comment]
I don't think so, you can add possibleNames on South Korea in country_utils.js
Korea, South then try again, I'm trying to improve this web service, I'm not development an application

**Comment by buster95** [comment]
I add `Korea, South` in possible names, this was results
image



This PR fix country data too for a lot of countries
With this PR
image

Without this PR
image

**Comment by EliteDaMyth** [comment]
update the pr wih your latest code.

**Comment by buster95** [comment]
I gonna do rollback for **`countries/{country name}`** endpoint but I think is a bad idea 🤷‍♂️

**Comment by buster95** [comment]
@EliteDaMyth finished, you can check this

File: server.js [hunk 1]
```diff
@@ -69,19 +70,38 @@ app.get("/historical/:country", async function (req, res) {
...
+  // const standardizedCountryName = countryMap.standardizeCountryName(req.params.country.toLowerCase());
+  // let country = countries.find(e => {
+  //   // see if strict was even a parameter
+  //   if (req.query.strict) {
+  //     return req.query.strict.toLowerCase() == 'true' ? e.country.toLowerCase() === standardizedCountryName : e.country.toLowerCase().includes(standardizedCountryName)
+  //   }
+  //   else {
+  //     return e.country.toLowerCase().includes(standardizedCountryName);
+  //   }
+  // });
```
Comment (ebwinters) [inline server.js]: Need to uncomment this for the endpoint to work correctly

File: server.js [hunk 2]
```diff
@@ -69,19 +70,38 @@ app.get("/historical/:country", async function (req, res) {
...
-      return e.country.toLowerCase().includes(standardizedCountryName);
+      return req.query.strict.toLowerCase() == 'true' ?
+        e.country.toLowerCase() === countryData.country.toLowerCase() :
+        e.country.toLowerCase().includes(countryData.country)
+    } else {
+      if (countryData.country) {
+        return e.country.toLowerCase().includes(countryData.country.toLowerCase());
+      }
     }
   });
```
Comment (ebwinters) [inline server.js]: Need to delete this

File: utils/country_utils.js [hunk 3]
```diff
@@ -1,566 +1,309 @@
...
+    { country: 'Barbados', iso2: 'BB', iso3: 'BRB', _id: 52, lat: 13.1667, long: -59.5333 },
+    { country: 'Belarus', iso2: 'BY', iso3: 'BLR', _id: 112, lat: 53, long: 28 },
+    { country: 'Belgium', iso2: 'BE', iso3: 'BEL', _id: 56, lat: 50.8333, long: 4 },
+    { country: 'Belize', iso2: 'BZ', iso3: 'BLZ', _id: 84, lat: 17.25, long: -88.75 },
+    { country: 'Benin', iso2: 'BJ', iso3: 'BEN', _id: 204, lat: 9.5, long: 2.25 },
+    { country: 'Bermuda', iso2: 'BM', iso3: 'BMU', _id: 60, lat: 32.3333, long: -64.75 },
+    { country: 'Bhutan', iso2: 'BT', iso3: 'BTN', _id: 64, lat: 27.5, long: 90.5 },
+    { country: 'Bolivia, Plurinational State of', iso2: 'BO', iso3: 'BOL', _id: 68, lat: -17, long: -65 },
+    { country: 'Bolivia', iso2: 'BO', iso3: 'BOL', _id: 68, lat: -17, long: -65 },
+    { country: 'Bosnia and Herzegovina', iso2: 'BA', iso3: 'BIH', _id: 70, lat: 44, long: 18 },
```
Comment (ebwinters) [inline utils/country_utils.js]: Change country name to Bosnia

File: utils/country_utils.js [hunk 4]
```diff
@@ -1,566 +1,309 @@
...
+    { country: 'Comoros', iso2: 'KM', iso3: 'COM', _id: 174, lat: -12.1667, long: 44.25 },
+    { country: 'Congo', iso2: 'CG', iso3: 'COG', _id: 178, lat: -1, long: 15 },
+    { country: 'Congo, the Democratic Republic of the', iso2: 'CD', iso3: 'COD', _id: 180, lat: 0, long: 25, possibleNames: ['DRC'] },
+    { country: 'Cook Islands', iso2: 'CK', iso3: 'COK', _id: 184, lat: -21.2333, long: -159.7667 },
+    { country: 'Costa Rica', iso2: 'CR', iso3: 'CRI', _id: 188, lat: 10, long: -84 },
+    { country: 'Coast D\'Ivoire', iso2: 'CI', iso3: 'CIV', _id: 384, lat: 8, long: -5, possibleNames: ['Ivory Coast'] },
+    { country: 'Croatia', iso2: 'HR', iso3: 'HRV', _id: 191, lat: 45.1667, long: 15.5 },
+    { country: 'Cuba', iso2: 'CU', iso3: 'CUB', _id: 192, lat: 21.5, long: -80 },
+    { country: 'Cyprus', iso2: 'CY', iso3: 'CYP', _id: 196, lat: 35, long: 33 },
+    { country: 'Czech Republic', iso2: 'CZ', iso3: 'CZE', _id: 203, lat: 49.75, long: 15.5, possibleNames: ['Czechia', 'Chequia'] },
```
Comment (ebwinters) [inline utils/country_utils.js]: Change country name to Czechia

File: utils/country_utils.js [hunk 5]
```diff
@@ -1,566 +1,309 @@
...
+    { country: 'Italy', iso2: 'IT', iso3: 'ITA', _id: 380, lat: 42.8333, long: 12.8333, possibleNames: ['Italia'] },
+    { country: 'Jamaica', iso2: 'JM', iso3: 'JAM', _id: 388, lat: 18.25, long: -77.5 },
+    { country: 'Japan', iso2: 'JP', iso3: 'JPN', _id: 392, lat: 36, long: 138, possibleNames: ['Japón'] },
+    { country: 'Jersey', iso2: 'JE', iso3: 'JEY', _id: 832, lat: 49.21, long: -2.13 },
+    { country: 'Jordan', iso2: 'JO', iso3: 'JOR', _id: 400, lat: 31, long: 36 },
+    { country: 'Kazakhstan', iso2: 'KZ', iso3: 'KAZ', _id: 398, lat: 48, long: 68 },
+    { country: 'Kenya', iso2: 'KE', iso3: 'KEN', _id: 404, lat: 1, long: 38 },
+    { country: 'Kiribati', iso2: 'KI', iso3: 'KIR', _id: 296, lat: 1.4167, long: 173 },
+    { country: 'North Korea', iso2: 'KP', iso3: 'PRK', _id: 408, lat: 40, long: 127, possibleNames: ['Korea del Norte', 'N. Korea', 'Korea, North', 'Democratic People\'s Republic of Korea'] },
+    { country: 'South Korea', iso2: 'KR', iso3: 'KOR', _id: 410, lat: 37, long: 127.5, possibleNames: ['Korea del Sur', 'S. Korea', 'Korea, South', 'Republic of Korea'] },
```
Comment (ebwinters) [inline utils/country_utils.js]: change country name to S. Korea

File: utils/country_utils.js [hunk 6]
```diff
@@ -1,566 +1,309 @@
...
+    { country: 'Tonga', iso2: 'TO', iso3: 'TON', _id: 776, lat: -20, long: -175 },
+    { country: 'Trinidad and Tobago', iso2: 'TT', iso3: 'TTO', _id: 780, lat: 11, long: -61, possibleNames: ['Trinidad & Tobago'] },
+    { country: 'Tunisia', iso2: 'TN', iso3: 'TUN', _id: 788, lat: 34, long: 9 },
+    { country: 'Turkey', iso2: 'TR', iso3: 'TUR', _id: 792, lat: 39, long: 35 },
+    { country: 'Turkmenistan', iso2: 'TM', iso3: 'TKM', _id: 795, lat: 40, long: 60 },
+    { country: 'Turks and Caicos Islands', iso2: 'TC', iso3: 'TCA', _id: 796, lat: 21.75, long: -71.5833, possibleNames: ['Turks and Caicos'] },
+    { country: 'Tuvalu', iso2: 'TV', iso3: 'TUV', _id: 798, lat: -8, long: 178 },
+    { country: 'Uganda', iso2: 'UG', iso3: 'UGA', _id: 800, lat: 1, long: 32 },
+    { country: 'Ukraine', iso2: 'UA', iso3: 'UKR', _id: 804, lat: 49, long: 32 },
+    { country: 'United Arab Emirates', iso2: 'AE', iso3: 'ARE', _id: 784, lat: 24, long: 54, possibleNames: ['uae'] },
```
Comment (ebwinters) [inline utils/country_utils.js]: Change country name to UAE

File: utils/country_utils.js [hunk 7]
```diff
@@ -1,566 +1,309 @@
...
+    { country: 'United Kingdom', iso2: 'GB', iso3: 'GBR', _id: 826, lat: 54, long: -2, possibleNames: ['UK'] },
```
Comment (ebwinters) [inline utils/country_utils.js]: change country name to UK

File: utils/country_utils.js [hunk 8]
```diff
@@ -1,566 +1,309 @@
...
+    { country: 'United States', iso2: 'US', iso3: 'USA', _id: 840, lat: 38, long: -97, possibleNames: ['USA', 'Estados Unidos'] },
```
Comment (ebwinters) [inline utils/country_utils.js]: Change country name to USA

File: server.js (hunk 2 above)
Comment (buster95) [reply server.js]: @ebwinters I can't delete this remember, country names has changed in redis storage, names are differents than world meter country names, therefore your code dosen't work

Comment (ebwinters) [reply server.js]: I tested locally and it worked fine

**Comment by EliteDaMyth** [comment]
the PR is conflicting.

**Comment by buster95** [comment]
@EliteDaMyth I was talking with @ebwinters in Discord, he will review my PR

File: server.js (hunk 2 above)
Comment (buster95) [reply server.js]: Your code work but not for all countries

**Pull request closed without merging**